source/vibe/core/parallelism.d

```diff
@@ -200,6 +201,29 @@ auto parallelMap(alias fun, R)(R items, shared(TaskPool) task_pool, ChannelConfi
 
 	static struct Result {
 		private State* state;
```

Comment:
Any particular disadvantages to using `std.typecons.RefCounted` that you have in mind?

Reply:
Could work here, I just generally stay away from it due to all the shortcomings it used to have (not sure about the status quo).

Reply:
Changed to use `RefCounted` and appears to work fine.